### DISCUSSION TITLE
Enabling mutation testing for Robolectric

### DISCUSSION THREAD

**Discussion by sktpy** [robolectric/robolectric#9040] [Ideas]
Aim of this discussion is to gather inputs and opinions regarding the [mutation testing](https://en.wikipedia.org/wiki/Mutation_testing) for Robolectric.

**Comment by utzcoz** [comment]
@sktpy Thanks for your idea. Can you expand more details/thoughts with your origin message? cc @hoisie.

**Comment by sktpy** [comment]
**Need:** Tests for the tests.

Mutation testing introduces small changes (mutator, eg: removing method call) in code resulting in the mutant, against which the existing test suite is run to check if it is failing (mutant is killed :heavy_check_mark: ), or passing (mutant survives :x:).
If our test suite still passes after the code is mutated, it means either we have missed some test case, or we have redundant/bad logic in our code.

Existing surviving mutations can be analysed to find the cause, and can be killed by either adding new test cases that will fail if the said code is mutated, or by changing the logic of the code. If both options are not feasible, it can remain suppressed.

The testing process can be automated by frameworks like [PIT](https://pitest.org/).
Mutation coverage goals can be set in CI for new PRs, to identify and act on surviving mutants before they get merged.
